decision-records/008-reset-the-javascript-api.md

```diff
@@ -0,0 +1,72 @@
+# Reset the JavaScript API for our components
+
+**Date:** 2023-07-20
+
+**Status:** Accepted
+
+## Decision
+
+In v4.x, for each component class, we will mark all class and instance properties other than the constructor and the `init` method as deprecated.
+
+In v5.0 we will then add JSDoc annotations to mark the deprecated properties as private. We will also annotate any 'internal' classes, like the `I18n` class, as internal.
```

Comment:
Probably going too far, but they're not always classes that are "internal only"

We haven't actually configured access for `Accordion.defaults` etc static properties (yet?)

````suggestion
In v5.0 we will then add JSDoc annotations to mark the deprecated properties as private.

Some component properties will remain publicly accessible but we'll annotate them for internal use only, such as:

```mjs
Accordion.defaults
new Accordion().config
new Accordion().i18n
```
````

Reply:
`Accordion.config` and `Accordion.i18n` are both currently private – why would we want to make them public?

Reply:
As in _literally_ public to a browser, we can't stop them being public

Just giving examples of three things that might be `@public`, `@protected` and `@internal`

From https://github.com/alphagov/govuk-design-system-architecture/pull/31/files#r1276470115 just need to widen "**'internal' classes**" to include all the other things (like helpers and utilities)

```suggestion
In v5.0 we will then add JSDoc annotations to mark the deprecated properties as private. We will also annotate any internal classes, helpers and utilities—like the `I18n` class—with JSDoc `@internal`.
```